picamera2/picamera2.py

```diff
@@ -534,8 +533,10 @@ def create_preview_configuration(self, main={}, lores=None, raw=None, transform=
             self.align_stream(lores, optimal=False)
         raw = self._make_initial_stream_config({"format": self.sensor_format, "size": main["size"]}, raw, self._raw_stream_ignore_list)
         # Let the framerate vary from 12fps to as fast as possible.
-        controls = {"NoiseReductionMode": libcamera.controls.draft.NoiseReductionModeEnum.Minimal,
-                    "FrameDurationLimits": (100, 83333)} | controls
+        ctrls = set([ctrl.name for ctrl in self.camera.controls.keys()])
+        if "NoiseReductionMode" in ctrls and "FrameDurationLimits" in ctrls:
+            controls = {"NoiseReductionMode": libcamera.controls.draft.NoiseReductionModeEnum.Minimal,
```

Comment:
Maybe
```
if "NoiseReductionMode" in self.camera_controls and "FrameDurationLimits" in self.camera_controls:
```
woudl be a bit easier to read? Ditto for the other two places below where we have this.